README.md

```diff
@@ -115,21 +115,10 @@ spec:
       WORKDIR simple-kmod
 
       # Prep and build the module
-      RUN make buildprep KVER=$(rpm -q --qf "%{VERSION}-%{RELEASE}.%{ARCH}"  kernel-core) KMODVER=${KMODVER} \
-      && make all       KVER=$(rpm -q --qf "%{VERSION}-%{RELEASE}.%{ARCH}"  kernel-core) KMODVER=${KMODVER} \
+      RUN make all       KVER=$(rpm -q --qf "%{VERSION}-%{RELEASE}.%{ARCH}"  kernel-core) KMODVER=${KMODVER} \
       && make install   KVER=$(rpm -q --qf "%{VERSION}-%{RELEASE}.%{ARCH}"  kernel-core) KMODVER=${KMODVER}
 
-      # Add the helper tools
-      WORKDIR /root/kvc-simple-kmod
-      ADD Makefile .
-      ADD simple-kmod-lib.sh .
-      ADD simple-kmod-wrapper.sh .
-      ADD simple-kmod.conf .
-      RUN mkdir -p /usr/lib/kvc/ \
-      && mkdir -p /etc/kvc/ \
-      && make install
-
-      RUN systemctl enable kmods-via-containers@simple-kmod
+      RUN echo "simple-kmod" > /etc/modules.d/simple-kmod.conf
```

Comment:
Why do we need this?

Reply:
I think this is what makes the node to automatically run `modprobe simple-kmod` on boot time (not sure though).
We can also change the command in the daemonset to `modprobe simple-kmod && /sbin/init` to achieve the same behaviour.

Reply:
That loads module at start time.

Reply:
How about we do not use `/sbin/init` and run `modprobe simple-kmod` as the command instead?

Reply:
```suggestion
      RUN make all install KVER=$(rpm -q --qf "%{VERSION}-%{RELEASE}.%{ARCH}" kernel-core) KMODVER=${KMODVER} && \
          echo "simple-kmod" > /etc/modules.d/simple-kmod.conf
```

Reply:
If we just run modprobe then writing to `/etc/modules.d/simple-kmod.conf` is not needed anymore

Reply:
Won't it collide with the lifecycle of running `modprobe -r simple-kmod` on daemon set removal?
I remember it from your demo of V2. I think doing what you are suggesting will not remove the module on cleanup.

Reply:
If not then I an in for removing `/sbin/init`

Reply:
maybe i am missing something, but why do we need to write into simple-kmod.conf or to run make all install?
the driver is supposed to be run loaded only in case driver container is running, and the modprobe command of the DS will always load the driver

Reply:
I don't see any `/etc/modules.d` inside DTK but rather `/etc/modules-load.d`.
Are you sure this is the correct path @enriquebelarte ?
```
[root@356f4d125961 simple-kmod]# echo "simple-kmod" > /etc/modules.d/simple-kmod.conf
bash: /etc/modules.d/simple-kmod.conf: No such file or directory
[root@356f4d125961 simple-kmod]# echo "simple-kmod" > /etc/modules-load.d/simple-kmod.conf
[root@356f4d125961 simple-kmod]#
```

Reply:
> we just run modprobe then writing to /etc/modules.d/simple-kmod.conf is not needed anymore

The issue with that is that the user can't override the `cmd` in the daemon set in that case, because if he does, the module won't be loaded anymore.

Reply:
@yevgeny-shnaidman 
make all - build the driver (gcc)
make install - create some links to make `modprobe` aware of `simple-kmod`
running `modprob simple-kmod` or writing to `simple-kmod.conf` is actually loading the driver.

The only downside of making the DS starging the driver (modprobe) is that the user will not be able to override that command but with the current approach he will be able to and the driver will still load.

Reply:
@ybettan i think that /etc/modules-load.d/ is used only during boot time, which is not relevant for our use case. And how is user supposed to override the command? user is the one that creates DS, so he defines the command to install the driver

Reply:
I don't think that running `/sbin/init` provides any value here.

Reply:
> i think that /etc/modules-load.d/ is used only during boot time, which is not relevant for our use case.

This explains why I didn't manage to make it work, still wondering how @enriquebelarte made it work though.

> And how is user supposed to override the command? user is the one that creates DS, so he defines the command to install the driver

Agree.

Reply:
We may need to replace `command: ["/sbin/init"]` with `command: ["modprob", "simple-kmod"]` in the DS section of the README.